src/core/logical_layers/renderers/BivariateRenderer.tsx

```diff
@@ -180,6 +181,7 @@ export class BivariateRenderer extends LogicalLayerDefaultRenderer {
         return;
       }
       const layer = { ...layerStyle, id: layerId, source: this._sourceId };
+      generatedMapboxLayersParents.set.dispatch(layerId, this.id);
       layerByOrder(map, this._layersOrderManager).addAboveLayerWithSameType(
```

Comment:
if it every time goes 2 calls `generatedMapboxLayersParents.set.dispatch `then `layerByOrder. addAboveLayerWithSameType `why not to put `generatedMapboxLayersParents.set` inside `addAboveLayerWithSameType` ?  🤔

Reply:
initially i didn't do it to specifically show that we create a bind with UI-layers and mapbox layers

i will think if i can optimize it, thanks!